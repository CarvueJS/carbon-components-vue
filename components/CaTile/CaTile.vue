<template>
  <div
    v-if="clickHandler"
    @click="clickHandler"
    data-tile="clickable"
    class="bx--tile bx--tile--clickable"
  >
    <slot></slot>
  </div>
  <div v-else data-tile="clickable" class="bx--tile">
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'ca-tile',
  props: {
    clickHandler: {
      type: Function,
      default: null,
    },
  },
};
</script>

<style></style>
