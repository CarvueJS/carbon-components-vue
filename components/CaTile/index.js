import CaTile from './CaTile.vue';

CaTile.install = (Vue) => {
  Vue.component(CaTile.name, CaTile);
};

export default CaTile;
