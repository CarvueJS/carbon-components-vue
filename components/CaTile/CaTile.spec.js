import { shallow } from 'vue-test-utils';
import CaTile from './CaTile.vue';

describe('CaTile', () => {
  test('is a Vue instance', () => {
    const wrapper = shallow(CaTile);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
})
