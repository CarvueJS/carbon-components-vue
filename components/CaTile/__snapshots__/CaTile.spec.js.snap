// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaTile Mathes snapshot 1`] = `<div data-tile="clickable" class="bx--tile"></div>`;
